core/templates/dev/head/components/CkEditorWidgetsGenerator.js

```diff
@@ -0,0 +1,190 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview dynamically generate CKEditor widgets for the rich text
+ components.
+ */
+
+oppia.run([
```

Comment:
I am concerned about this. If I'm not mistaken there can only be one run() method per angular app, so it is a bit like oppia.config() in app.js. No one who edits app.js is going to realize that there's a run() method already defined in this file, so this is going to make the codebase less maintainable. This was less of a problem when run() was in app.js because everyone can see it and add to the run() method as needed. But now that it's in a separate place altogether I think the maintainability concerns are real.

Can the code for setting up CKEditor instead be defined in a service that the CKEditorRteDirective depends on (and throws an error at initialization if the service is not available)?

/cc @bansalnitish

Reply:
From this [source](https://stackoverflow.com/a/29528572/3688189) it seems to me that there can be multiple `oppia.run`s.

Reply:
Ah, my bad, then. Thanks for finding that, @vojtechjelinek!

In that case I withdraw my concern. However, I think we should have a standard convention for naming all .run() files, similar to what we have for directives, components, etc.

@ankita240796 @vojtechjelinek any thoughts here? ...Runner.js perhaps? Or ...Initializer.js (since run() is supposed to execute at the start)?

Reply:
`Initializer.js` sounds better to me.